src/docs/arch-glossary.dox

```diff
@@ -4,31 +4,97 @@ WiredTiger has a lot of domain specific nomenclature - this page attempts
+<tr><td>raw<td>schema<td>data as it appears on-disk, typically in a packed format.
+<tr><td>read timestamp<td>transactions<td>a timestamp that specifies what data should be visible.
+<tr><td>recno<td>general<td>a 64 bit integer. When used as a key, the Btree is said to be a column store, and the keys are assigned incrementally by WiredTiger when new records are inserted.
+<tr><td>reconcile<td>general<td>convert the in-memory version of a page to a more compact form for file storage.
+<tr><td>recover<td>general<td>a procedure used on opening WiredTiger whereby, starting at the last checkpoint written, log records are played back to bring the database up to date.
+<tr><td>ref<td>btree<td>see page reference
+<tr><td>RLE<td>general<td>see run length encoding
+<tr><td>run length encoding<td>general<td>also known as "RLE".  A technique to save space whereby repeated items can be represented by a count followed by the item.  For example the string of letters \sic "AAAAAAABBBCDDDD" might be encoded as \sic "7A3BC4D" using "out of band" digits.
+<tr><td>salvage<td>general<td>functionality to repair Btree disk files.
+<tr><td>schema<td>schema<td>the algorithms that use metadata to map the raw keys and values in Btrees to higher level abstractions like tables with typed columns.
+<tr><td>session<td>general<td>an encapsulation of information stored for single thread of control in the application. This encapsulation is kept in a handle that is used in the API.  Sessions manage transactions and performs the transactional operations.
+<tr><td>skip list<td>general<td>a variation of a sorted linked list that allows for faster (O(log(N))) searching. It is used in WiredTiger as a lock-free data structure for keys inserted on a page.
+<tr><td>spin lock<td>general<td>a simple locking algorithm that continually attempts to acquire a lock, appropriate for resources that are likely to be available and may be held for a short time.  This lock is CPU intensive while it is waiting.  Contrast with mutex
+<tr><td>split<td>btree<td>see page split
+<tr><td>stable timestamp<td>transactions<td>a connection-wide value, checkpoints will not include commits newer than this value.
+<tr><td>table<td>schema<td>an abstraction that allows one or more btrees to be used in a coordinated way, which is specified by the schema.  The table coordinates indices and column groups.
+<tr><td>thread<td>general<td>a thread of control.  WiredTiger uses pthreads for POSIX systems.
+<tr><td>thread group<td>general<td>a group of threads that work together.  Eviction is performed by a thread group.
+<tr><td>timestamp<td>transactions<td>refers to one of several types of 64 bit values used to control transactional visibility of data in WiredTiger.  See oldest timestamp, stable timestamp, commit timestamp, read timestamp.
+<tr><td>transaction<td>transactions<td>a unit of work performed in WiredTiger that is atomic and consistent with the specified isolation levels, and durability levels.
```

Comment:
Remove final comma. Also, I'd use the singular for level. E.g., A txn only has one isolation level.